sw/device/lib/testing/test_framework/check.h

```diff
@@ -157,6 +157,25 @@
     }                                                                          \
   } while (false)
 
+/**
+ * Checks the characters of two strings are the same,
+ * up to and including the first null character.
+ * The CHECK macro is called on each character pair.
+ *
+ * @param actual_ The first string in the comparison.
+ * @param expected_ The second string in the comparison.
+ */
+#define CHECK_STR_EQ(actual_, expected_)                                     \
+  do {                                                                       \
+    size_t i = 0;                                                            \
+    const char *expected = (expected_);                                      \
+    const char *actual = (actual_);                                          \
+    do {                                                                     \
+      CHECK(actual[i] == expected[i], "\"%s\" != \"%s\"", actual, expected); \
```

Comment:
It could be handy to mention the first mismatching byte's position, especially for large strings.

Reply:
Great idea! I've added it.